ui/app/routes/vault/cluster/secrets/backend/configuration/index.js

```diff
@@ -5,28 +5,132 @@
       const canRead = await this.store
-        .findRecord('capabilities', `${backend.id}/config`)
+        .findRecord('capabilities', `${secretEngineModel.id}/config`)
         .then((response) => response.canRead);
       // only set these config params if they can read the config endpoint.
       if (canRead) {
         // design wants specific default to show that can't be set in the model
-        backend.casRequired = backend.casRequired ? backend.casRequired : 'False';
-        backend.deleteVersionAfter = backend.deleteVersionAfter ? backend.deleteVersionAfter : 'Never delete';
+        secretEngineModel.casRequired = secretEngineModel.casRequired
+          ? secretEngineModel.casRequired
+          : 'False';
+        secretEngineModel.deleteVersionAfter = secretEngineModel.deleteVersionAfter
+          ? secretEngineModel.deleteVersionAfter
+          : 'Never delete';
       } else {
         // remove the default values from the model if they don't have read access otherwise it will display the defaults even if they've been set (because they error on returning config data)
-        backend.set('casRequired', null);
-        backend.set('deleteVersionAfter', null);
-        backend.set('maxVersions', null);
+        secretEngineModel.set('casRequired', null);
+        secretEngineModel.set('deleteVersionAfter', null);
+        secretEngineModel.set('maxVersions', null);
+      }
+    }
+    // If the engine is configurable fetch the config model(s) for the engine and return it alongside the model
+    if (CONFIGURABLE_SECRET_ENGINES.includes(secretEngineModel.type)) {
+      let configModels = await this.fetchConfig(secretEngineModel.type, secretEngineModel.id);
+      configModels = this.standardizeConfigModels(configModels);
+
+      return hash({
+        secretEngineModel,
+        ...configModels,
+      });
+    }
+    return hash({ secretEngineModel });
+  }
+
+  standardizeConfigModels(configModels) {
+    // standardize the configModels to an array so that the component can handle it correctly
+    Array.isArray(configModels) ? configModels : (configModels = [configModels]);
+    // make sure no items in the array are null or undefined
+    configModels.forEach((configModel) => {
+      if (!configModel) {
+        configModels.splice(configModels.indexOf(configModel), 1);
       }
+    });
+
+    return configModels;
+  }
+
+  async fetchConfig(type, id) {
+    switch (type) {
+      case 'aws':
+        return await this.fetchAwsConfigs(id);
+      case 'ssh':
+        return await this.fetchSshCaConfig(id);
     }
-    return backend;
+  }
+
+  async fetchAwsConfigs(id) {
+    // AWS has two configuration endpoints root and lease, return an array of these responses.
+    const configArray = [];
+    const configRoot = await this.fetchAwsRoot(id);
+    const configLease = await this.fetchAwsLease(id);
+    configArray.push(configRoot, configLease);
+    return configArray;
+  }
+
+  async fetchAwsLease(id) {
```

Comment:
since this method and the next one are very similar, I wonder if they could be combined into a `fetchAwsConfig(id, modelType)` method which does the handling consistently but uses the passed `modelType` instead of the hardcoded one. Would make this just a little more DRY and since this route has a lot of methods, I think that might be valuable